### PR TITLE
Prevent Gray Header bar from affecting header region of flexible layout template.

### LINKED ISSUE #5149: Fixed style issues with Flexible layout template UI and make it more consistent with the layout/blocks UI
This is part of the #4046 meta.

During this week's design meeting, we discussed various styling issues with the flexible layout templates UI:

1. The "Header" region has a darker gray background than the rest of the regions, and when you hover over it, there is a dark gray bottom border shown:

    Screenshot 2021-08-12 at 23 52 21

   ...for context, hovering over the other regions makes them acquire a white background/container, with a bottom/right border of the same dark gray color as the one that appears when hover :

    image

   It seems that the header inherits the style of the header of the admin theme 🤷🏼 

2. The next issue with this UI is that there is no indication that the various rows can be dragged, unless you hover over the rows (so users without a pointing device won't know). The blocks in the layout admin UI on the other hand have a crosshair icon shown always at their top-left.

   Since rows in flexy templates can only be moved up/down, we should add an up/down arrow icon.

3. If you configure a row in a flexy template, and change the "Row width behavior" (under the "Style" fieldset) setting to "full width", then the region style changes to indicate that, but it may not be as clear that this is the case. Compare the widths of the containers of the "CONTENT" vs the "FOOTER" rows in the screenshot below:

   Screenshot 2021-08-13 at 00 20 16

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/5149

### CODE REVIEW

File: core/modules/layout/css/layout.admin.css [hunk 1]
```diff
@@ -173,7 +180,7 @@
 .layout-editor-region-title .dropbutton-wrapper {
   float: left; /* LTR */
   width: 140px;
-  margin: 4px 0 10px
+  margin: 4px 00;
```
Comment (klonos) [inline core/modules/layout/css/layout.admin.css]: ??

File: core/modules/layout/css/layout.flexible.admin.css [hunk 2]
```diff
@@ -0,0 +1,82 @@
+/**
+ * Flexible template admin css.
+ */
+
+ .layout-flexible-region-top .label {
+   display: inline-block;
+   margin: 0;
+   padding: 4px 12px 4px 0; /* LTR */
```
Comment (klonos) [inline core/modules/layout/css/layout.flexible.admin.css]: There's no matching `[dir="rtl"]` for this

File: core/modules/layout/css/layout.admin.css (hunk 1 above)
Comment (jenlampton) [reply core/modules/layout/css/layout.admin.css]: oops, fixed! (I was trying to add the `;`)

File: core/modules/layout/css/layout.flexible.admin.css (hunk 2 above)
Comment (jenlampton) [reply core/modules/layout/css/layout.flexible.admin.css]: added

File: core/modules/layout/css/layout.admin.css [hunk 3]
```diff
@@ -153,14 +157,17 @@
...
+}
 .layout-editor-region-title .label {
   float: left; /* LTR */
   margin: 0;
   padding: 4px 12px 4px 4px; /* LTR */
   line-height: 1.2;
   text-transform: uppercase;
   font-weight: normal;
-  color: #908990;
+  color: #5b575b; /* WCAG AAA on white bg */
```
Comment (klonos) [inline core/modules/layout/css/layout.admin.css]: As previously agreed, remove this, and add a GitHub comment.

```suggestion
  color: #5b575b;
```

File: core/modules/layout/css/layout.admin.css [hunk 4]
```diff
@@ -173,7 +180,7 @@
 .layout-editor-region-title .dropbutton-wrapper {
   float: left; /* LTR */
   width: 140px;
-  margin: 4px 0 10px
+  margin: 4px 0 10px;
```
Comment (klonos) [inline core/modules/layout/css/layout.admin.css:183]: Good catch 👍🏼 ...how did we miss that? ...cant wait to have a linter as part of our PR tests.

File: core/modules/layout/layout.flexible.inc [hunk 5]
```diff
@@ -115,6 +115,11 @@ function layout_flexible_template_configure_form($form, &$form_state, LayoutFlex
   form_load_include($form_state, 'inc', 'layout', 'layout.admin');
   form_load_include($form_state, 'inc', 'layout', 'layout.flexible');
 
+  $form['help'] = array(
+    '#type' => 'help',
+    '#markup' => t('This page allows you to add or remove rows, and change the row order by dragging them up and down. You can also change the number of regions within each row, and adjust the width of each region.'),
```
Comment (klonos) [inline core/modules/layout/layout.flexible.inc]:
```suggestion
    '#markup' => t('This page allows you to add or remove rows, and change the order of rows by dragging them up and down. You can also change the number of regions within each row, and adjust the width of each region.'),
```

**Review by klonos — APPROVED** [review]
LGTM 👍🏼

**Comment by quicksketch** [comment]
Merged into 1.x and 1.20.x. Thanks!

**Pull request merged**